relnotes/update.feature.md

```diff
@@ -0,0 +1,10 @@
+### Neo Update request
+
+`dhtproto.client.DhtClient`, `dhtproto.client.request.Update`
+
+The new request, `Update`, fetches a record value, allows the user to specify an
+updated value, and replaces the old value in the node with the new value. Note
+that the request will notice if the value being updated has been modified by
+another request, while the Update is in progress. If this happens, the Update
+will be rejected (the client should retry the request).
```

Comment:
Feels like the `{De}SerializeMethod` stuff should be discussed here (not necessarily in detail, but the principle of what is available).

Reply:
Seems a bit specific to me. My feeling is that details of usage should be (and are) covered by the usage examples.

Reply:
TBH I am a little bit mystified as to why swarm/dhtproto are even caring about (de)serialization.  I suppose it makes it simpler to define the delegates without having to provide them a manually-defined context where deserialization functionality is available.  But OTOH it takes us away from the historical assumption that it's for the user to care about what values are, not the swarm protocols.

Is there something specific to the update request that requires this?

Reply:
Nope, the protocol doesn't care. It's just a convenience for the user, who would have to implement this anyway. The other DHT neo requests have the same feature.